### ISSUE TITLE
Stack manipulation issues

### ISSUE THREAD

**Issue by catwell** [shmul/lightningmdb#12]
You have several invalid operations from a stack point of view in this library. You can see it if you use it with a Lua built with `-DLUA_USE_APICHECK`.

For instance your `*_register` functions look like this:

```lua
void env_register(lua_State* L) {
  luaL_newmetatable(L,ENV);           -- +1
  lua_set_funcs(L,ENV,env_methods);   -- 0
  lua_settable(L,-1);                 -- -2

  luaL_getmetatable(L,ENV);           -- +1
  lua_setfield(L,-1,"__index");       -- -1
}
```

I have added the stack changes as comments on the side. As you can see it is wrong at `lua_settable` except if you have a string at the top of the stack before calling the function, which is not the case.

I suppose what you want is something like this (I don't really understand what you were trying to achieve with `lua_settable`):

```lua
void env_register(lua_State* L) {
  luaL_newmetatable(L,ENV); 
  lua_set_funcs(L,ENV,env_methods);

  luaL_getmetatable(L,ENV);
  lua_setfield(L,-1,"__index");
  lua_pop(L, 1);
}
```

On 5.1 I think you also have an issue with the call to `lua_setglobal` in `lua_set_funcs`.

**Comment by shmul** [comment]
Thanks for that. I will add the build directive and will address this issue.

**Comment by shmul** [comment]
I just published version 0.9.18.2-1 with a fix to the problem. I hope I got it right.
Thanks for taking the time to report.

Cheers,
Shmul
